src/main/java/baritone/pathing/movement/movements/MovementDescend.java

```diff
@@ -69,18 +85,26 @@ public double calculateCost(CalculationContext context) {
         return ImmutableSet.of(src, dest.up(), dest);
     }
 
-    public static void cost(CalculationContext context, int x, int y, int z, int destX, int destZ, MutableMoveResult res) {
+    public static void cost(CalculationContext context, int x, int y, int z, int destX, int destZ, MutableMoveResult res, EnumFacing direction) {
         double totalCost = 0;
         IBlockState destDown = context.get(destX, y - 1, destZ);
-        totalCost += MovementHelper.getMiningDurationTicks(context, destX, y - 1, destZ, destDown, false);
+        totalCost += MovementHelper.getMiningDurationTicks(context, destX, y - 1, destZ, destDown, false, SpaceRequest.withLowerPlayerSpace(SpaceRequest.fromFaces(direction.getOpposite())));
+        if (totalCost >= COST_INF) {
+            return;
+        }
+        totalCost += MovementHelper.getMiningDurationTicks(context, destX, y, destZ, false, SpaceRequest.withAllPlayerSpace(SpaceRequest.fromFaces(direction.getOpposite())));
+        if (totalCost >= COST_INF) {
+            return;
+        }
+        totalCost += MovementHelper.getMiningDurationTicks(context, destX, y + 1, destZ, true, SpaceRequest.withUpperPlayerSpace(SpaceRequest.fromFaces(direction.getOpposite()))); // only the top block in the 3 we need to mine needs to consider the falling blocks above
         if (totalCost >= COST_INF) {
             return;
         }
-        totalCost += MovementHelper.getMiningDurationTicks(context, destX, y, destZ, false);
+        totalCost += MovementHelper.getMiningDurationTicks(context, x, y, z, false, SpaceRequest.fromFaces(direction));
```

Comment:
Sorry. We just can't do this. This double counts ALL cost.

If there's a traverse for example deep underground, this would count the cost of breaking the starting position (two blocks) and the cost of breaking the destination position (two blocks). That just isn't correct. The core movement logic assumes that by the time a movement starts, the previous movement is completed and has moved the player to the start of this movement, which includes breaking any blocks that may have been in the way. This breaks that assumption for the sole benefit of allowing us to open doors that we're standing in at the beginning of the path. There are better ways to accomplish that.